### PR TITLE
feature/4: Added support for more types

### LINKED ISSUE #4: Add support for more types
Currently only very few parquet types are supported.

Todo: add support for others

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #4

### CODE REVIEW

File: src/test/java/com/exasol/edmlgenerator/parquet/ParquetColumnToMappingDefinitionConverterTest.java [hunk 1]
```diff
@@ -47,4 +110,96 @@ private void assertConvertsToToDecimalMapping(final Type type, final int scale,
...
+                .convert(logicalTimestampType);
+        assertAll(//
+                () -> assertThat(toTimestampMapping.getDestinationName(), equalTo("MY_TIMESTAMP")), //
+                () -> assertThat(toTimestampMapping.isUseTimestampWithLocalTimezoneType(), equalTo(true))//
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("testConvertToStringCases")
+    void testConvertToString(final LogicalTypeAnnotation annotation) {
```
Comment (kaklakariada) [inline src/test/java/com/exasol/edmlgenerator/parquet/ParquetColumnToMappingDefinitionConverterTest.java:147]: Put testConvertToStringCases next to this method to make the code easier to read.

File: src/test/java/com/exasol/edmlgenerator/parquet/ParquetColumnToMappingDefinitionConverterTest.java [hunk 2]
```diff
@@ -47,4 +110,96 @@ private void assertConvertsToToDecimalMapping(final Type type, final int scale,
...
+                () -> assertThat(toVarcharMapping.getVarcharColumnSize(), equalTo(expectedSize)),
+                () -> assertThat(toVarcharMapping.getDestinationName(), equalTo(expectedName))//
+        );
+    }
+
+    @Test
+    void testConvertList() {
+        final Type itemType = Types.primitive(INT32, REQUIRED).named("element");
+        final GroupType listType = Types.list(REQUIRED).element(itemType).named("ids");
+        System.out.println(listType.toString());
```
Comment (kaklakariada) [inline src/test/java/com/exasol/edmlgenerator/parquet/ParquetColumnToMappingDefinitionConverterTest.java:165]:
```suggestion
```

File: src/test/java/com/exasol/edmlgenerator/parquet/ParquetColumnToMappingDefinitionConverterTest.java [hunk 3]
```diff
@@ -47,4 +110,96 @@ private void assertConvertsToToDecimalMapping(final Type type, final int scale,
...
+        assertConvertsToJsonMapping(listType, "IDS");
+    }
+
+    @Test
+    void testConvertMap() {
+        final Type keyType = Types.primitive(BINARY, REQUIRED).as(LogicalTypeAnnotation.stringType()).named("key");
+        final Type valueType = Types.primitive(INT32, REQUIRED).named("value");
+        final GroupType listType = Types.map(REQUIRED).key(keyType).value(valueType).named("scores");
+        System.out.println(listType.toString());
```
Comment (kaklakariada) [inline src/test/java/com/exasol/edmlgenerator/parquet/ParquetColumnToMappingDefinitionConverterTest.java:174]:
```suggestion
```

File: src/test/java/com/exasol/edmlgenerator/parquet/ParquetColumnToMappingDefinitionConverterTest.java [hunk 4]
```diff
@@ -47,4 +110,96 @@ private void assertConvertsToToDecimalMapping(final Type type, final int scale,
...
+        assertConvertsToJsonMapping(listType, "SCORES");
+    }
+
+    @Test
+    void testConvertEnum() {
+        final Type keyType = Types.primitive(BINARY, REQUIRED).as(LogicalTypeAnnotation.stringType()).named("key");
+        final Type valueType = Types.primitive(INT32, REQUIRED).named("value");
+        final GroupType listType = Types.map(REQUIRED).key(keyType).value(valueType).named("scores");
+        System.out.println(listType.toString());
```
Comment (kaklakariada) [inline src/test/java/com/exasol/edmlgenerator/parquet/ParquetColumnToMappingDefinitionConverterTest.java:183]:
```suggestion
```

File: src/test/java/com/exasol/edmlgenerator/parquet/ParquetColumnToMappingDefinitionConverterTest.java [hunk 5]
```diff
@@ -47,4 +110,96 @@ private void assertConvertsToToDecimalMapping(final Type type, final int scale,
...
+                .convert(listType);
+        assertAll(//
+                () -> assertThat(toJsonMapping.getVarcharColumnSize(), equalTo(2_000_000)),
+                () -> assertThat(toJsonMapping.getDestinationName(), equalTo(expectedName))//
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("getUnsupportedTypes")
+    void testUnsupported(final Type unsupportedType) {
```
Comment (kaklakariada) [inline src/test/java/com/exasol/edmlgenerator/parquet/ParquetColumnToMappingDefinitionConverterTest.java:198]: Put `getUnsupportedTypes` next to this test

File: src/main/java/com/exasol/edmlgenerator/parquet/ParquetColumnToMappingDefinitionConverter.java [hunk 6]
```diff
@@ -15,33 +15,54 @@
...
+        } else {
+            final NonLogicalTypeConvertVisitor visitor = new NonLogicalTypeConvertVisitor();
+            column.accept(visitor);
+            return visitor.getResult();
+        }
     }
 
-    private static class ConvertVisitor
-            implements TypeVisitor, LogicalTypeAnnotation.LogicalTypeAnnotationVisitor<Void> {
+    private static class NonLogicalTypeConvertVisitor implements TypeVisitor {
```
Comment (kaklakariada) [inline src/main/java/com/exasol/edmlgenerator/parquet/ParquetColumnToMappingDefinitionConverter.java:57]: Inner classes are quite large. Maybe extract to separate file?

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: src/test/java/com/exasol/edmlgenerator/parquet/ParquetColumnToMappingDefinitionConverterTest.java (hunk 1 above)
Comment (jakobbraun) [reply src/test/java/com/exasol/edmlgenerator/parquet/ParquetColumnToMappingDefinitionConverterTest.java:147]: Unfortunately the formatter moves static methods to the top. :-/

File: src/test/java/com/exasol/edmlgenerator/parquet/ParquetColumnToMappingDefinitionConverterTest.java (hunk 5 above)
Comment (jakobbraun) [reply src/test/java/com/exasol/edmlgenerator/parquet/ParquetColumnToMappingDefinitionConverterTest.java:195]: same

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**